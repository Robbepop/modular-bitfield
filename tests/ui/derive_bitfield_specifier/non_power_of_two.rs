// Bitfield enums with a number of variants other than a power of two should
// fail to compile.
//
// (Or, if you implemented the optional #[bits = N] enum approach mentioned in
// the explanation of test case 06, then enums with non-power-of-two variants
// without a #[bits = N] attribute should fail to compile.)

use modular_bitfield::prelude::*;

#[derive(BitfieldSpecifier)]
pub enum Bad {
    Zero,
    One,
    Two,
}

fn main() {}
