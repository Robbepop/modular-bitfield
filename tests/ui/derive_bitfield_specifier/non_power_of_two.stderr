error: BitfieldSpecifier expected a number of variants which is a power of 2, specify #[bits = 2] if that was your intent
  --> tests/ui/derive_bitfield_specifier/non_power_of_two.rs:11:1
   |
11 | pub enum Bad {
   | ^^^
